pkg/controllers/dnspolicy/dns_helper.go

```diff
@@ -453,11 +454,11 @@ func (dh *dnsHelper) getDNSHealthCheckProbes(ctx context.Context, gateway *gatew
 	})
 }
 
-func getProbesForEndpoint(endpoint *v1alpha1.Endpoint, probes []*v1alpha1.DNSHealthCheckProbe) []*v1alpha1.DNSHealthCheckProbe {
+func getProbesForEndpoint(endpoint *v1alpha1.Endpoint, probes []*v1alpha1.DNSHealthCheckProbe, policyName, listenerName string) []*v1alpha1.DNSHealthCheckProbe {
 	retProbes := []*v1alpha1.DNSHealthCheckProbe{}
 	for _, probe := range probes {
 		for _, target := range endpoint.Targets {
-			if strings.Contains(probe.Name, target) {
+			if endpointHealthCheckProbe(target, policyName, listenerName, probe.Name) {
```

Comment:
`endpointHealthCheckProbe` seems oddly named, would probably be clearer what is happening if you just did the check for equality here:
```
dnsHealthCheckProbeName(target, policyName, listenerName) == probe.Name
```

Reply:
yep. makes sense. addressed in the latest push. 👍